plugins/inputs/snmp_trap/snmp_trap_test.go

```diff
@@ -408,11 +416,13 @@ func TestReceiveTrap(t *testing.T) {
 				testutil.MustMetric(
 					"snmp_trap", // name
 					map[string]string{ // tags
-						"oid":     ".1.3.6.1.6.3.1.1.5.1",
-						"name":    "coldStart",
-						"mib":     "SNMPv2-MIB",
-						"version": "3",
-						"source":  "127.0.0.1",
+						"oid":          ".1.3.6.1.6.3.1.1.5.1",
+						"name":         "coldStart",
+						"mib":          "SNMPv2-MIB",
+						"version":      "3",
+						"source":       "127.0.0.1",
+						"context_name": "foo_context_name",
+						"engine_id":    "6261725f656e67696e655f6964",
```

Comment:
The `engine_id` looks high cardinality, is it different for every remote agent?  I don't know too much about how this is used, is it really important to be able to lookup on this value and if not perhaps we should skip adding it.

Reply:
Engine ID is unique per remote agent and never changes, even if an attribute like the IP or MAC address changes.  I think context name is meant to be human readable but not change often.  For cardinality to become a problem with either you'd have to have as many unique remote agents sending traps as your cardinality ceiling.

I added them because they were available and seemed useful.  If you're querying v3 traps from a specific remote agent but its IP address isn't static, engine_id would be the best way to get what you want.

I'm open to removing either or both of them or adding settings so that they're disabled by default if you're concerned about cardinality.

Reply:
From your description of engine_id I think it will be fine.